cmd/helm/template_test.go

```diff
@@ -94,6 +98,11 @@ func TestTemplateCmd(t *testing.T) {
 			cmd:    fmt.Sprintf("template '%s' --show-only templates/service.yaml", chartPath),
 			golden: "output/template-show-only-one.txt",
 		},
+		{
+			name:   "template with show-only one and output-dir",
+			cmd:    fmt.Sprintf("template '%s' --show-only templates/service.yaml --output-dir '%s'", chartPath, outputDir),
```

Comment:
This unit test yields empty output and fails, yet the command does not yield an error. I tried the exact same command from my CLI and the expected output can be seen. To be investigated ...

Output from my CLI:
~~~
[akaris@linux helm]$ ../../bin/helm template 'testdata/testcharts/subchart' --show-only templates/service.yaml --output-dir '/tmp/template-with-show-only-one-and-output-dir' 
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/templates/service.yaml

[akaris@linux helm]$ find /tmp/template-with-show-only-one-and-output-dir
/tmp/template-with-show-only-one-and-output-dir
/tmp/template-with-show-only-one-and-output-dir/subchart
/tmp/template-with-show-only-one-and-output-dir/subchart/templates
/tmp/template-with-show-only-one-and-output-dir/subchart/templates/service.yaml
~~~

Unit test output:
~~~
=== RUN   TestTemplateCmd/template_with_show-only_one_and_output-dir
    helm_test.go:61: running cmd (attempt 1): template 'testdata/testcharts/subchart' --show-only templates/service.yaml --output-dir '/tmp/template-with-show-only-one-and-output-dir'
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/templates/service.yaml
    helm_test.go:67: does not match golden file testdata/output/template-show-only-one-and-output-dir.txt
        
        WANT:
        'wrote /tmp/template-with-show-only-one-and-output-dir/subchart/templates/service.yaml
        
        '
        
        GOT:
        '
        '
~~~

No idea why.

Reply:
The current implementations of`writeToFile` write to STDOUT via `fmt` instead of using cobra's output stream. This is why this test fails.  That's an issue with the current implementation and way beyond the scope of this fix (maybe something for a decent refactor).
The fmt.Println can be seen inside the test's output, though.

Meanwhile, I fixed the test with:
~~~
echo '' > ./cmd/helm/testdata/output/template-show-only-one-and-output-dir.txt
~~~

Even though this will only test for an empty result, it does catch the error message that was seen before.

I ran the new unit test against current upstream and got:
~~~
=== RUN   TestTemplateCmd/template_with_show-only_one_and_output-dir
    helm_test.go:61: running cmd (attempt 1): template 'testdata/testcharts/subchart' --show-only templates/service.yaml --output-dir '/tmp/template-with-show-only-one-and-output-dir'
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/templates/subdir/serviceaccount.yaml
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/templates/subdir/role.yaml
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/templates/subdir/rolebinding.yaml
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/charts/subcharta/templates/service.yaml
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/charts/subchartb/templates/service.yaml
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/templates/service.yaml
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/templates/tests/test-config.yaml
wrote /tmp/template-with-show-only-one-and-output-dir/subchart/templates/tests/test-nothing.yaml
    helm_test.go:64: expected error, got 'could not find template templates/service.yaml in chart'
    helm_test.go:67: does not match golden file testdata/output/template-show-only-one-and-output-dir.txt
        
        WANT:
        '
        '
        
        GOT:
        'Error: could not find template templates/service.yaml in chart
        '
~~~